include/deal.II/numerics/tensor_product_matrix_creator.h

```diff
@@ -379,6 +531,270 @@ namespace TensorProductMatrixCreator
       fe, quadrature, boundary_ids, cell_extent, n_overlap);
   }
 
+  template <typename Number>
+  FullMatrix<Number>
+  create_1d_cell_mass_matrix(const FiniteElement<1>     &fe,
+                             const Number               &h,
+                             const std::pair<bool, bool> include_endpoints,
+                             std::vector<unsigned int>   numbering)
+  {
+    if (dynamic_cast<const FE_DGQ<1> *>(&fe) == nullptr &&
+        numbering.size() == 0)
+      {
+        Assert(
+          include_endpoints.first == true && include_endpoints.second == true,
+          ExcMessage(
+            "You tried to generate a 1D mass matrix with excluding boundary "
+            "dofs for a non-DGQ element without providing a numbering."));
+      }
+
+    if (numbering.size() == 0)
+      {
+        numbering.resize(fe.dofs_per_cell);
+        std::iota(numbering.begin(), numbering.end(), 0);
+      }
+
+    const unsigned int degree          = fe.degree;
+    const unsigned int n_dofs_per_cell = fe.dofs_per_cell;
+    const Number      &JxW             = h;
```

Comment:
`JxW` is used for the jacobian times the weight. This is just the jacobian alone. The variable name should be changed.

Reply:
variable removed I used (h * quadrature.weight(q) ) in the integration loop